ggshield/sca/file_selection.py

```diff
@@ -68,17 +69,30 @@ def is_not_excluded_from_sca(scannable: Scannable) -> bool:
     return not any(part in SCA_IGNORE_LIST for part in scannable.path.parts)
 
 
-def tar_sca_files_from_git_repo(directory: Path, ref: str, client: SCAClient) -> bytes:
+def tar_sca_files_from_git_repo(
+    directory: Path,
+    ref: str,
+    client: SCAClient,
+    exclusion_regexes: Optional[Set[re.Pattern]] = None,
+) -> bytes:
     """Builds a tar containing SCA files from the git repository at
     the given directory, for the given ref. Empty string denotes selection
     from staging area."""
-    # TODO: add exclusion patterns
+    if exclusion_regexes is None:
+        exclusion_regexes = set()
+
     if ref == INDEX_REF:
         all_files = get_staged_filepaths(wd=str(directory))
     else:
         all_files = get_filepaths_from_ref(ref, wd=str(directory))
 
-    sca_files_result = client.compute_sca_files(files=[str(path) for path in all_files])
+    sca_files_result = client.compute_sca_files(
```

Comment:
Maybe we should add a shortcut in `compute_sca_files` to avoid an API call in case the arg is an empty list, WDYT?